test/test_driver/test_sq_store.py

```diff
@@ -1,13 +1,31 @@
 from __future__ import annotations
 
 import random
-from typing import TYPE_CHECKING
+import pytest
+import tempfile
 
 from squirrel.driver import StoreDriver
 from squirrel.iterstream import FilePathGenerator
+from squirrel.serialization import MessagepackSerializer
+from squirrel.store import SquirrelStore
 
-if TYPE_CHECKING:
-    from squirrel.store import SquirrelStore
+
+def test_store_creation(dummy_sq_store: SquirrelStore, num_samples: int) -> None:
+    """Test store creation with non empty path."""
+    uncleaned_store = SquirrelStore(url=dummy_sq_store.url, serializer=MessagepackSerializer(), clean=False)
+    assert len(list(uncleaned_store.keys())) == num_samples
+
+    cleaned_store = SquirrelStore(url=dummy_sq_store.url, serializer=MessagepackSerializer(), clean=True)
+    assert not cleaned_store.fs.exists(dummy_sq_store.url)
+    assert len(list(cleaned_store.keys())) == 0
+
+
+@pytest.mark.parametrize("clean", [True, False])
+def test_store_creation_empty_path(test_path: str, clean: bool) -> None:
+    """Test store creation with an empty path."""
+    store = SquirrelStore(url=test_path, serializer=MessagepackSerializer(), clean=clean)
```

Comment:
In case we change how we implement `test_path`, e.g. we return an existing path.

After importing
```
from squirrel.fsspec.fs import get_fs_from_url
```

```suggestion
    fs = get_fs_from_url(path)
    assert not fs.exists(test_path)
    store = SquirrelStore(url=test_path, serializer=MessagepackSerializer(), clean=clean)
```